test/recipes/02-test_windows_registry.t

```diff
@@ -0,0 +1,42 @@
+#! /usr/bin/env perl
+# Copyright 2015-2021 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use OpenSSL::Test;
+use OpenSSL::Test::Utils;
+use OpenSSL::Test qw/:DEFAULT bldtop_file data_file srctop_file cmdstr/;
+
+setup("test_windows_registry");
+
+plan skip_all => "Windows registry tests are only available on windows"
+    if $^O !~ /(MSWin)/;
+
+plan tests => 6;
+
+ok(run(cmd(["reg.exe", "add", "HKLM\\SOFTWARE\\WOW6432Node\\OpenSSL", "/t", "REG_EXPAND_SZ", "/v", "OPENSSLDIR", "/d", "TESTOPENSSLDIR", "/f"])),
+   "Adding OPENSSLDIR registry key");
+
+ok(run(cmd(["reg.exe", "add", "HKLM\\SOFTWARE\\WOW6432Node\\OpenSSL", "/t", "REG_EXPAND_SZ", "/v", "ENGINESDIR", "/d", "TESTENGINESDIR", "/f"])),
+   "Adding ENGINESDIR registry key");
+
+ok(run(cmd(["reg.exe", "add", "HKLM\\SOFTWARE\\WOW6432Node\\OpenSSL", "/t", "REG_EXPAND_SZ", "/v", "MODULESDIR", "/d", "TESTMODULESDIR", "/f"])),
+   "Adding MODULESDIR registry key");
```

Comment:
Hmmm.....so this is going to overwrite any existing registry entries?

Reply:
yes, do you think we need to restore them afterwards?

Reply:
Possibly skip the test if they already exist - and if we do run the test, then delete them afterwards

Reply:
This is bad. Although I understand we want to test it, running `make test` should not make any system-wide changes. Can we test this in a github CI job instead?

Reply:
well, sure, its getting tested in CI now.  But testing it implies that the registry needs to have the keys set.  I suppose the most appropriate solution is to do what @mattcaswell  suggests and only preform the test if the keys aren't already set, then delete them afterward.  That will have the odd side effect of skipping the test on any windows platform that has run the installer already of course, but perhaps thats ok, as having run the installer implies that their not building from source there anyway

Reply:
We could do that, but in that event the test will be skipped on local systems in which the registry keys aren't already set, I suppose that might be ok though, but its not ideal.

Reply:
Really, the make test is not supposed to mess up with your system. And I assume if you're an unprivileged user trying to run make test, this test would fail because you should not be able to edit the system wide registry as an unprivileged user. Unless there is some kind of local user override for the SOFTWARE keys - I have no idea if there is such thing not being a Windows developer for a very long time.